examples/qr_codes.rs

```diff
@@ -0,0 +1,33 @@
+//! Demonstrate encoding with uppercase characters suitable for QR codes.
+
+use std::fmt;
+
+use bech32::primitives::hrp;
+use bech32::Fe32;
+
+fn main() {
+    let hrp = hrp::BC;
+    let witness_version = Fe32::Q; // Segwit Version 0.
+    let witness_program = [
+        0x75, 0x1e, 0x76, 0xe8, 0x19, 0x91, 0x96, 0xd4, 0x54, 0x94, 0x1c, 0x45, 0xd1, 0xb3, 0xa3,
+        0x23, 0xf1, 0x43, 0x3b, 0xd6,
+    ];
+
+    let mut address = String::new();
+    let mut writer = UpperWriter(&mut address);
+    bech32::segwit::encode_to_fmt_unchecked(&mut writer, &hrp, witness_version, &witness_program)
+        .expect("failed to encode address to QR code");
+
+    println!("\n\nExample Bitcoin QR code URI: bitcoin:{}", address);
+}
+
+struct UpperWriter<W: fmt::Write>(W);
+
+impl<W: fmt::Write> fmt::Write for UpperWriter<W> {
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        for c in s.chars() {
+            self.0.write_char(c.to_ascii_uppercase())?;
+        }
+        Ok(())
+    }
+}
```

Comment:
In 047a29e89017a5a4a4f4ccdbdec9bbad69c7f11c:

We should just have an uppercase version of `encode_to_fmt`. This code is complicated and requires allocations when there would otherwise be none.